src/test/java/seedu/sudohr/logic/commands/AddEmployeeToDepartmentCommandTest.java

```diff
@@ -0,0 +1,294 @@
+package seedu.sudohr.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.sudohr.testutil.Assert.assertThrows;
+
+import java.nio.file.Path;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.sudohr.commons.core.GuiSettings;
+import seedu.sudohr.logic.commands.exceptions.CommandException;
+import seedu.sudohr.model.Model;
+import seedu.sudohr.model.ReadOnlySudoHr;
+import seedu.sudohr.model.ReadOnlyUserPrefs;
+import seedu.sudohr.model.SudoHr;
+import seedu.sudohr.model.department.Department;
+import seedu.sudohr.model.department.DepartmentName;
+import seedu.sudohr.model.employee.Employee;
+import seedu.sudohr.model.employee.Id;
+import seedu.sudohr.testutil.TypicalDepartmentNames;
+import seedu.sudohr.testutil.TypicalEmployees;
+
+public class AddEmployeeToDepartmentCommandTest {
+
+    // Handle adding an employee
+    @Test
+    public void execute_employeeAcceptedByDepartment_addSuccessful() throws CommandException {
+
+        ModelStubAcceptingEmployeeAdded modelStub = new ModelStubAcceptingEmployeeAdded();
+        modelStub.addEmployee(TypicalEmployees.ALICE);
+        Department toAdd = new Department(TypicalDepartmentNames.DEPARTMENT_NAME_FIRST);
+        modelStub.addDepartment(toAdd);
+
+        CommandResult commandResult = new AddEmployeeToDepartmentCommand(new Id("101"),
+                TypicalDepartmentNames.DEPARTMENT_NAME_FIRST).execute(modelStub);
+
+        assertEquals(String.format(AddEmployeeToDepartmentCommand.MESSAGE_ADD_EMPLOYEE_TO_DEPARTMENT_SUCCESS,
+                TypicalEmployees.ALICE, TypicalDepartmentNames.DEPARTMENT_NAME_FIRST),
+                commandResult.getFeedbackToUser());
+        assertTrue(modelStub.sudoHr.getDepartment(TypicalDepartmentNames.DEPARTMENT_NAME_FIRST)
+                .hasEmployee(TypicalEmployees.ALICE));
+    }
+    // Handle adding duplicate employee
+    @Test
+    public void execute_duplicateEmployeeInDepartment_throwsCommandException() throws CommandException {
```

Comment:
What is the expected behavior if i re-add Alice into the department? Think it would throw duplicateEmployee error as well, which is desired.

Just a minor point, at this stage of adding employee into departments, all employees should be valid and not have any clashing identities fields so perhaps it is more reasonable to test adding the same person object to department. But nothing wrong with this testing on id field uh, just that it raises eyebrows why not test on email and phone fields

Reply:
they are done in JsonSerializableSudoHrTest

Reply:
Discussed: can test for each of the primary keys - id, email, phone number for defensive programming